### PR TITLE
[dash] Refactor libsai

### PR DESCRIPTION
**Pull request by kcudnik**
Many changes, added dummy apis for support swss/orchagent start

* Add dummy SAI base APIs
* Start using ObjectIdManager in DashSai
* Add ObjectIdManager class
* Extract switch apis to separate file
* Extract port apis to separate file
* Add dash quad macros for quad SAI api
* Dash SAI add quad generic api skeleton
* Add Makefile include files dependencies
* Move out non template files to src directory
* Move generate_dash_api.sh script to proper Makefile
* Refactor style for saiapi template and add logs

### CODE REVIEW

**Comment by kcudnik** [comment]
it should be better to review one commit at a time

File: dash-pipeline/tests/libsai/vnet_out/vnet_out.cpp [hunk 1]
```diff
@@ -48,7 +60,7 @@ int main(int argc, char **argv)
     QUERY_STATUS_CHECK(status, SAI_API_DASH_ENI);
 
     sai_direction_lookup_entry_t dle = {};
-    dle.switch_id = switch_id;
+    dle.switch_id = 0; // TODO switch_id, it fails when switch id is non zero
```
Comment (kcudnik) [inline dash-pipeline/tests/libsai/vnet_out/vnet_out.cpp:63]: @marian-pritsak @chrispsommers, in this PR i added check, that switch must be created first before creating any other object, but somehow this fails if switch_id is non zero, mutate table entry fails, this comes from deep bmv2/p4 internals, not sure why this happens

Comment (chrispsommers) [reply dash-pipeline/tests/libsai/vnet_out/vnet_out.cpp:63]: @kcudnik It'd be helpful to include a console dump of bmv2 model for this failure.

Comment (kcudnik) [reply dash-pipeline/tests/libsai/vnet_out/vnet_out.cpp:63]: hmm, locally it passes, but here it failed: https://github.com/sonic-net/DASH/actions/runs/6174596066/job/16759546592?pr=438#step:13:25

Comment (kcudnik) [reply dash-pipeline/tests/libsai/vnet_out/vnet_out.cpp]: testing again

Comment (kcudnik) [reply dash-pipeline/tests/libsai/vnet_out/vnet_out.cpp]: false alarm

**Comment by chrispsommers** [comment]
@kcudnik FYI I am working my way through this one commit at a time, I have not forgotten.

File: dash-pipeline/SAI/src/objectidmanager.h [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+            /**
+             * @brief Set of allocated switch indexes.
+             */
+            std::set<uint32_t> m_switchIndexes;
+
+            /**
+             * @brief Object index indexer map.
+             *
+             * Each obejct type have it's own starting index from zero for easy
```
Comment (chrispsommers) [inline dash-pipeline/SAI/src/objectidmanager.h:129]: s/obejct/object/

s/it's/its/ (appears 3 places in this diff) - weird rules of English :)

File: dash-pipeline/SAI/Makefile [hunk 3]
```diff
@@ -1,19 +1,6 @@
...
-	    -lgrpc++ \
-	    -lgrpc \
-	    -lpiall \
-	    -lpi_dummy \
-	    -lpthread \
-	    -labsl_synchronization \
-	    -g
-
 
+all:
```
Comment (chrispsommers) [inline dash-pipeline/SAI/Makefile]: I like this change, it's long overdue, The old make content was terribly obsolete, leftover from early days. Thanks!

**Review by chrispsommers — CHANGES_REQUESTED** [review]
Minor typos, otherwise excellent PR. I was able to build and test it locally, it's great.

File: dash-pipeline/SAI/src/objectidmanager.h (hunk 2 above)
Comment (kcudnik) [reply dash-pipeline/SAI/src/objectidmanager.h:129]: English is not my first language :D

Comment (kcudnik) [reply dash-pipeline/SAI/src/objectidmanager.h]: addressed

**Review by chrispsommers — APPROVED** [review]

**Pull request merged**